lib/src/serialization/serializer.dart

```diff
@@ -72,7 +82,11 @@ class Serializer extends Module {
                   clk, reset: reset, en: latchInput, deserialized.elements[i])
               : deserialized.elements[i]);
     }
-    serialized <= dataOutput.elements.selectIndex(count);
+    if (deserialized.dimensions.length > 1) {
```

Comment:
aren't these two statements identical?  why have the `if` at all?

Reply:
Good catch.  I had moved the addOutputArray down there while experimenting, only to discover that I could use the same name for the output after constructing Logic or LogicArray.